op-devstack/dsl/operator_fee.go

```diff
@@ -0,0 +1,167 @@
+package dsl
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack/match"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/txintent/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/txintent/contractio"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type OperatorFee struct {
+	commonImpl
+
+	l1Client     *L1ELNode
+	l2Network    *L2Network
+	systemConfig bindings.SystemConfig
+	l1Block      bindings.L1Block
+
+	originalScalar   uint32
+	originalConstant uint64
+}
+
+type OperatorFeeValidationResult struct {
+	TransactionReceipt   *types.Receipt
+	ExpectedOperatorFee  *big.Int
+	ActualTotalFee       *big.Int
+	VaultBalanceIncrease *big.Int
+}
+
+func NewOperatorFee(t devtest.T, l2Network *L2Network, l1EL *L1ELNode) *OperatorFee {
+	systemConfig := bindings.NewBindings[bindings.SystemConfig](
+		bindings.WithClient(l1EL.EthClient()),
+		bindings.WithTo(l2Network.Escape().Deployment().SystemConfigProxyAddr()),
+		bindings.WithTest(t))
+
+	l1Block := bindings.NewBindings[bindings.L1Block](
+		bindings.WithClient(l2Network.inner.L2ELNode(match.FirstL2EL).EthClient()),
+		bindings.WithTo(predeploys.L1BlockAddr),
+		bindings.WithTest(t))
+
+	originalScalar, err := contractio.Read(systemConfig.OperatorFeeScalar(), t.Ctx())
+	t.Require().NoError(err)
+	originalConstant, err := contractio.Read(systemConfig.OperatorFeeConstant(), t.Ctx())
+	t.Require().NoError(err)
+
+	return &OperatorFee{
+		commonImpl:       commonFromT(t),
+		l1Client:         l1EL,
+		l2Network:        l2Network,
+		systemConfig:     systemConfig,
+		l1Block:          l1Block,
+		originalScalar:   originalScalar,
+		originalConstant: originalConstant,
+	}
+}
+
+func (of *OperatorFee) checkCompatibility() bool {
+	_, err := contractio.Read(of.systemConfig.OperatorFeeScalar(), of.ctx)
+	if err != nil {
+		of.t.Skipf("Operator fee methods not available in devstack: %v", err)
+		return false
+	}
+	return true
+}
+
+func (of *OperatorFee) skipIfIncompatible() {
+	if !of.checkCompatibility() {
+		of.t.Skip("Operator fee methods not available")
+	}
+}
+
+func (of *OperatorFee) GetSystemOwner() *EOA {
+	systemOwnerKey := devkeys.SystemConfigOwner.Key(of.l2Network.ChainID().ToBig())
+	return NewKey(of.t, of.l2Network.Escape().Keys().Secret(systemOwnerKey)).User(of.l1Client)
+}
+
+func (of *OperatorFee) SetOperatorFee(scalar uint32, constant uint64) {
+	of.skipIfIncompatible()
```

Comment:
Do we need these calls in each function, or could it perhaps just be in the constructor?